### ISSUE TITLE
Show number of violations in addon-a11y tab

### ISSUE THREAD

**Issue by cocco3** [storybookjs/storybook#16235]
**Is your feature request related to a problem? Please describe**
When using the a11y addon, it's not apparent if a story has any accessibility violations unless you open the "Accessibility" tab. While we encourage developers to verify no issues, people often forget and things get missed.

**Describe the solution you'd like**
Show the number of violations in the tab, similar to Controls. This could also be an opt-in feature so it doesn't disrupt anyone's existing UI. It could look like this:

<img width="468" alt="Screen Shot 2021-10-04 at 5 32 55 PM" src="https://user-images.githubusercontent.com/1775017/135943056-65a061b7-6803-4673-ba18-9d0400109966.png">

**Describe alternatives you've considered**
n/a

**Are you able to assist to bring the feature to reality?**
No

**Comment by shilman** [comment]
Love this feature! ❤️  cc @MichaelArestad

**Comment by mikaoelitiana** [comment]
Hi there! I am interested to work on this feature request

**Comment by jsomsanith** [comment]
Hi 👋 
Don't know if this feature is still wanted, but I made the PR #17983 to display the number of a11y issues in the tab title.

But that won't be sufficient for your initial need. The axe scan runs only if the tab is active. But if you don't go in the accessibility tab, it doesn't run so no number displayed.
This is done on purpose because on complicated stories, the scan can take some time and blocks the thread, making storybook unresponsive during the scan.

Would an option to run it with inactive tab be ok in your case ? The default behaviour stays as today, with no run.
We should then think about different titles in case of
- no scan result yet
- no issue in the scan

**Comment by MichaelArestad** [comment]
@jsomsanith Thank you for creating a PR!

> But that won't be sufficient for your initial need. The axe scan runs only if the tab is active. But if you don't go in the accessibility tab, it doesn't run so no number displayed.

That's okay for now. I would love it if the addon ran even if the tab wasn't active so a user would have an idea of how many violations exist before they even open the addon. That said, there are technical challenges to that which I leave to engineers.

> Would an option to run it with inactive tab be ok in your case ?

An option could work, but I wonder if we could do something that doesn't require opt-in. @shilman what do you think?

**Comment by MichaelArestad** [comment]
> no scan result yet

I don't think we need to show anything here if it hasn't run yet.

In the scenario with no confirmed failures:

<img width="333" alt="image" src="https://user-images.githubusercontent.com/1123119/163879221-be9a1d27-e902-40d9-94fb-1cc73864ecb2.png">

**Comment by jsomsanith** [comment]
> That's okay for now. I would love it if the addon ran even if the tab wasn't active so a user would have an idea of how many violations exist before they even open the addon. That said, there are technical challenges to that which I leave to engineers.

Yeah, as I detailed in my previous message the scan can take a lot of time, making storybook unresponsive :/ 
I made the PR some years ago to run it only on active tab.
Technically, we can't run it in a web worker to free the main thread because we won't have access to the DOM.

In term of UX, if we want to run the scan at story render even with inactive addon-a11y tab, we have those choices:
- adding the addon and have potentially frozen times like what I experienced before
- adding the addon, it doesn't run at story render, but we add an option to do that, making it clear in doc that it can freeze on stories with complex html markup. 

The first solution will first make user think that storybook is not performant or doesn't work properly IMO
The second one is harder to use because it needs more configuration.

**Comment by jsomsanith** [comment]
But either way, agree with you that displaying the number of issues in tab when we have the scan results is ok for now. It's a first step and we dig a little more to see how we can reintroduce the auto scan.
Will you review me PR @MichaelArestad ?

**Comment by jsomsanith** [comment]
PR merged, let's close the issue

**Comment by shilman** [comment]
Great Caesar's ghost!! I just released https://github.com/storybookjs/storybook/releases/tag/v6.5.0-beta.4 containing PR #17983 that references this issue. Upgrade today to the `@next` NPM tag to try it out!

```
npx sb upgrade --prerelease
```
